src/python/pants/backend/python/target_types_rules.py

```diff
@@ -182,6 +185,57 @@ async def generate_targets_from_python_test_utils(
     )
 
 
+class GenerateTargetsFromPexBinariesFromEntryPoints(GenerateTargetsRequest):
+    generate_from = PexBinariesFromEntryPointsGeneratorTarget
+
+
+@rule
+async def generate_targets_from_pex_binaries_from_entry_points(
+    request: GenerateTargetsFromPexBinariesFromEntryPoints,
+    union_membership: UnionMembership,
+) -> GeneratedTargets:
+    generator_addr = request.generator.address
+    entry_points_field = request.generator[PexEntryPointsField].value or []
+    overrides = request.generator[OverridesField].flatten()
+
+    # @TODO: Check for overlap:
+    #   E.g. `path.to.app` specd as well as `app`, that should error
+    #   do this by resolving entry points before creating binaries so all are of the form
+    #   `path.to.app`. (But note that the override field might have `app`)
+
+    def create_pex_binary(entry_point_spec: str) -> PexBinary:
+        target_fields = {
+            field.alias: field.value
+            for field in request.generator.field_values.values()
+            if not isinstance(field, (PexEntryPointsField, OverridesField))
+        }
+        target_fields.update(overrides.pop(entry_point_spec, {}))
+
+        return PexBinary(
+            {PexEntryPointField.alias: entry_point_spec, **target_fields},
+            # TOREVIEW: We Should discuss the address, because users wil likely want to package or
+            #   run these scripts, so the address they use shouldn't be too weird.
+            # E.g. path/to/scriptdir:generator_name#foo.par:baz is kinda lengthy
+            # ":" is a forbidden character in target names
+            generator_addr.create_generated(entry_point_spec.replace(":", "-")),
```

Comment:
Oh shoot, that's weird. I can't think of a better solution though.

Reply:
Not until the generators-aren't-targets proposal is in

Reply:
I unresolved this discussion. Curious if others have any ideas for a better delimiter than `-`. I think it makes sense, `app.py-main`. It works because Python wouldn't allow you to use `-` for a Python file name.

Reply:
Yes, we could definitely allow it. What do you think @thejcannon and others? I'm not sure if it risks being too confusing.

Reply:
The end-result is less confusing than the current impl IMO.

Reply:
Yeah, let's not have more than one `:` in a target address.

Reply:
To be clear, it sounds like @thejcannon and @kaos are advocating allowing `:` in the generated portion, and you're saying it's a bad idea @benjyw, right? I'm uncertain, but lean towards it being a bad idea.

Reply:
An alternative I might've had originally is if the user doesn't like the default generated one, they can `override` the `name` field (which would become the right-hand-side of the `#` probably, or just replace the name outright)

Reply:
cc @stuhood - thoughts on the `name`?

Reply:
I tend to see the target address a bit like an url.. and in that regard, having multiple separators in different parts is not a strange thing.. `target://src/python/proj:bin#cli.py:main` (where the `target://` part is made up, naturally, vs `https://some.domain:8443/resource/path#with/fragment?bin=entry:point` for comparison example.
What IS required for this to work, though, is very well defined parts. Where the URL is broken down into `<protocol>://<server>/<uri>` so perhaps it is such a well defined spec of what a target address is that needs to be put down on paper in a rather formal prose? ;)

Reply:
I attempted to define that spec in https://www.pantsbuild.org/docs/targets#target-addresses and the table at https://www.pantsbuild.org/docs/targets#target-generation. 

It's already pretty complex. I've advocated in https://github.com/pantsbuild/pants/issues/12917 that we should get rid of "file address syntax" and consolidate around the `#` syntax used for non-first-party targets to simplify things and make it more predictable. But Stu and Benjy have raised good points about preferring file-address syntax.

So I'm nervous to make address syntax even more complex than it is now. Although I recognize that `app.py:main` becoming `app.py-main` is not very predictable, either.

Reply:
Yeah, those are good for teaching about target addresses, I'm advocating writing a form of "RFC" style document that describes in detail the valid syntax for target addresses. With one of those in hand, you shouldn't have to be nervous about making changes when the syntax is well defined, including any corner cases.

I'd be happy to assist flesh one out, if it turns out to be a desirable thing to have for target adresses. :)

Reply:
(regardless of the acceptance of `:` in the generated name part or not :P )

Reply:
> I think it makes sense, `app.py-main`. It works because Python wouldn't allow you to use `-` for a Python file name.

This seems fine to me. I agree that we should not be changing the limitations on the address syntax right now, as there are other inbound changes (beyond the `#` syntax), and the complexity is already high. If in the future we do support `:` in the name, we wouldn't be forced to change this position, so the consequences of that open question are low.